File: sinapse_frontend/src/services/DeckService.ts
export class DeckService {
  async createDeck(userId: string, deckName: string) {
    const response = await fetch("http://127.0.0.1:8080/decks", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ user_id: userId, name: deckName }),
    });
    const data = await response.json();
    return data;
  }
  async getDecks(userId: string) {
    const response = await fetch(`http://127.0.0.1:8080/decks/${userId}`);
    const data = await response.json();
    return data;
  }
  async getDeck(id: string) {
    const response = await fetch(`http://127.0.0.1:8080/deck/${id}`);
    const data = await response.json();
    return data;
  }
  async deleteDeck(id: string) {
    const response = await fetch(`http://127.0.0.1:8080/decks/${id}`, {
      method: "DELETE",
    });
    const data = await response.json();
    return data;
  }
}

File: sinapse_bot/cogs/services/deck_service.py
from typing import List, Dict

import requests

from cogs.classes.flashcard import Flashcard



class DeckService:
    async def get_decks(self, user_id: str) -> List[Dict[str, str]]:
        response = requests.get(f"http://127.0.0.1:8080/decks/{user_id}")
        data = response.json()
        return data

    async def get_deck(self, id: str) -> List[Flashcard]:
        response = requests.get(f"http://127.0.0.1:8080/deck/{id}")
        data = response.json()
        flashcards = []
        for flashcard in data:
            flashcards.append(Flashcard(**flashcard))
        return flashcards